### ISSUE TITLE
TM1914 support?

### ISSUE THREAD

**Issue by NelloCris** [FastLED/FastLED#787]
Hi all, I would like to know if this IC TM1914 is supported from FastLed library (platform Arduino).
If not, someone have the datasheet in English of this IC? Thanks in advance.

N.

**Comment by BYohann** [comment]
Hello All, i would also like to know. and if rgbww or rgb+cct are supported, they are becoming standards for home lighting projects where RGB is not usefull.

**Comment by focalintent** [comment]
Rgbw support will be coming as part of a nearly full rewrite of the library.  I will make an announcement on the project homepage and the Reddit when it’s available (I don’t have a time frame for it yet)

**Comment by BYohann** [comment]
It sounds great, i hope it can released this year with roadmap. thank you for your work

**Comment by grego87** [comment]
> It sounds great, i hope it can released this year with roadmap. thank you for your work

Unfortunately [there are some bad news](https://github.com/FastLED/FastLED/issues/579#issuecomment-534396376).

**Comment by oraclesod** [comment]
Super sorry to read about the loss of one of the community, that super sad :( 

Is anyone still working on developing FastLed ? I also have the TM1914 chipset on my strip and was wondering if it would be possible to add support for this ?

Im more than happy to provide someone with a TM1914 strip etc if someone is able to help code the library for its support, i wish i could do it myself, i can code but not a this level...

I would swap the strip out for something else, but this is the only strip i could find that was addressable RGB with W/WW also on the same strip, i use a ESP8266 with ESPhome to control the W/WW perfectly, but have to use a crappy SP108E with its original firmware to control the colour side and the nasty app that doesn't allow much customization....

Or if someone can point me in the direction to whats needed to make this library work I'm happy to try to self teach myself.. i just dont know where to start...

**Comment by NonaSuomy** [comment]
Did these get supported TM1914 RGB+CW+WW? 

https://www.alibaba.com/product-detail/TM1914-RGBww-5in1-addressable-LED-Strip_60702633541.html

https://www.alibaba.com/product-detail/RGBWCCT-addressable-24V-5in1-RGB-CCT_60718722774.html

Thank you.

**Comment by marmilicious** [comment]
FastLED only supports RGB at this time.

**Comment by zackees** [comment]
Closing as a **duplicate** — this thread contains two separate asks, and both are tracked in better places now. Answering each, since neither ever got a current answer here.

## 1. TM1914 support — duplicate of #681

Still not supported, and the blocker is specific: the TM1914 drives an **inverted** line. `FLIP` exists as a template parameter on every clockless controller but is honored by exactly **one** backend in the tree — nRF52 PWM. Everywhere else it's accepted and ignored, and the newer `fl::channels` path has no polarity concept at all. Adding a TM1914 controller today would emit *non-inverted* data on nearly every platform while looking supported.

There's a second requirement on top: the TM1914 needs 6 bytes of frame-control data prepended to every frame, which the WS2812-style encoders have no notion of.

Full analysis is on **#681** (deferred, with the polarity plumbing tracked on **#8**). Please follow those rather than this thread.

## 2. RGBW / RGBWW — @marmilicious's 2022 answer is now out of date

> FastLED only supports RGB at this time.

That was correct then. It isn't now.

**RGBW is first-class.** There's an `fl::Rgbw` config type, a `.setRgbw()` builder on the controller, and **WS2814** shipped as a dedicated RGBW chipset (#3769). White-channel handling includes color-temperature configuration and LUT interpolation modes (`src/fl/gfx/rgbw.h`).

**RGBWW (5-channel RGB + warm-white + cool-white) also landed**, via the Channels API — #2558, "5-channel RGB+WW support via Channels API", is closed, and there's a working `examples/RGBWW/RGBWW.ino`:

```cpp
// ChannelOptions.mWhiteCfg is a fl::variant<fl::Empty, Rgbw, Rgbww>.
// Assigning Rgbww selects the 5-channel alternative.
opts.mWhiteCfg = fl::Rgbww( ... );
```

@oraclesod, @NonaSuomy, @BYohann — this is the RGB+CCT / 5-in-1 capability you were asking about, and the example is the place to start.

One caveat so nobody is misled: the **legacy** `CLEDController::setRgbww()` entry point still carries a doc comment saying *"Driver-side support arrives in later phases of the RGBWW work"* (`src/fl/channels/cled_controller.h:122`). Given #2558 is closed and the example exists, that comment looks stale — but use the **Channels API path shown in `examples/RGBWW`** rather than the legacy setter, which is the route that's actually exercised.

---

@oraclesod — your offer to donate a TM1914 strip stood unanswered for five years, and that's the kind of thing that would genuinely unblock the work. If you're still around and still willing, say so on #681, which is where the TM1914 work will happen.

Closing here so the two asks live where they're tracked: **#681** for TM1914, **#2558** / `examples/RGBWW` for 5-channel white.
